context-free.md

```diff
@@ -2,21 +2,31 @@
 
 ##  Who is this programming language for?
 
+This programming language seems to be primarily for graphic designers (either professionally or casually) who have some experience writing code and thinking mathematically. I could also see mathematicians who focus on geometry using it to play around with an idea, since it easily represents fractal images that would be hard to represent manually with something like TIKz.
 
 ## What is easy to do in this language? Why is it easy?
 
+This language makes it very easy to represent very complicated and abstract shapes. For example, you can represent an "infinite" amount of shapes with just a few lines. Actually having to specify everything in the image would be incredibly hard, but Context Free allows you to recursively specify shapes with properties as the reccurence relation. 
 
 ## What is hard to do in this language? Why is it hard?
 
+There were a few issues I ran into with the language. 
+ - First, I found it was occasionally hard to figure out what coordinates my images were centered around. Since the image window changes to fit your image, sometimes its hard to tell where the origin is. 
+ - Additionally, the syntax error messages weren't very descriptive, so debugging in this language isn't easy.
 
 ## How did you learn how to program in this language?
 
+I attempted to learn by looking at the documentation, but I found that a lot of the commands being abbreviated inconsistently made it hard to udnerstand. Instead, I looked at examples and tutorials that I found online until I figured out what worked and what didn't. 
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+It seems like the process has 2 steps. First, when you decide to render the program you write, the IDE for Context Free renders the image in the side window. This is analagous to just running a code script and looking at the results. Second, when you're satisfied, a "completed program" can have its rendered image saved. 
 
 ## What do you think is interesting about the ContextFree program you wrote?
+ 
+The thing I like about the program I wrote is its simplicity. The actual lines of code are quite simple, there are two main recursive shaps I defined, and I just put 8 of them in the image at various angles. However, the actual process of creating the image was just a series of repeatedly changing the hue and dimension and angle settings until I found things that I liked. So the program itself was always quite short, but went through many different configurations, all of which looked very different, until I settled on a theme. I think that's what I like about COntext Free, it lets you make very cool, unique images with very few lines of actual code. 
```

Comment:
I liked this about yours as well. My favorite parts were the elegance in the code and also the four subsections of colors (i.e. four "quarters") within the shape. Really, really cool and looks nice on the black as well.